### PR TITLE
Update build scripts

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR updates the build scripts to be consistent with RAJA. They needed some work after recent TOSS4 updates.
- Also updated RAJA TPL so host-configs match.

These scripts are not used in CI, but they have been tested manually on LC systems.

### CODE REVIEW

File: scripts/lc-builds/corona_sycl.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+cmake \
+  -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
+  -DSYCL_LIB_PATH:STRING="${SYCL_PATH}/lib" \
+  -C ${RAJA_HOSTCONFIG} \
+  -DENABLE_OPENMP=Off \
+  -DENABLE_CUDA=Off \
+  -DRAJA_ENABLE_TARGET_OPENMP=Off \
+  -DENABLE_ALL_WARNINGS=Off \
+  -DRAJA_ENABLE_SYCL=On \
```
Comment (MrBurmark) [inline scripts/lc-builds/corona_sycl.sh:50]: Should this just be ENABLE_SYCL?

Comment (rhornung67) [reply scripts/lc-builds/corona_sycl.sh:50]: Yes, it should. Thank you.

File: scripts/lc-builds/toss4_amdclang.sh [hunk 2]
```diff
@@ -53,23 +59,28 @@ module load cmake/3.23.1
...
-  -DGPU_TARGETS="${COMP_ARCH}" \
-  -DAMDGPU_TARGETS="${COMP_ARCH}" \
+  -DROCM_ROOT_DIR="${ROCM_PATH}" \
+  -DHIP_ROOT_DIR="${ROCM_PATH}/hip" \
+  -DHIP_PATH=${ROCM_PATH}/llvm/bin \
+  -DCMAKE_C_COMPILER=${ROCM_PATH}/llvm/bin/amdclang \
+  -DCMAKE_CXX_COMPILER=${ROCM_PATH}/llvm/bin/amdclang++ \
+  -DCMAKE_HIP_ARCHITECTURES="${COMP_ARCH}:xnack+" \
+  -DGPU_TARGETS="${COMP_ARCH}:xnack+" \
+  -DAMDGPU_TARGETS="${COMP_ARCH}:xnack+" \
```
Comment (MrBurmark) [inline scripts/lc-builds/toss4_amdclang.sh:79]: Why are we forcing xnack+ here? Are we trying to allow people to use address sanitizer?

File: scripts/lc-builds/toss4_clang.sh [hunk 3]
```diff
@@ -7,18 +7,18 @@
 # SPDX-License-Identifier: (BSD-3-Clause)
 ###############################################################################
 
-if [[ $# -lt 1 ]]; then
+if [ "$1" == "" ]; then
```
Comment (MrBurmark) [inline scripts/lc-builds/toss4_clang.sh:10]: Why change this, it was consistent with other uses before?

File: scripts/lc-builds/toss4_amdclang.sh (hunk 2 above)
Comment (rhornung67) [reply scripts/lc-builds/toss4_amdclang.sh]: That's a cut & paste error. Fixed.

File: scripts/lc-builds/toss4_clang.sh (hunk 3 above)
Comment (rhornung67) [reply scripts/lc-builds/toss4_clang.sh:10]: I think there's a mix of both checks. I'll take a pass through and make them all the same.

Comment (rhornung67) [reply scripts/lc-builds/toss4_clang.sh]: All scripts are consistent now.

File: scripts/lc-builds/corona_sycl.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  -C ${RAJA_HOSTCONFIG} \
+  -DENABLE_OPENMP=Off \
+  -DENABLE_CUDA=Off \
+  -DRAJA_ENABLE_TARGET_OPENMP=Off \
+  -DENABLE_ALL_WARNINGS=Off \
+  -DENABLE_SYCL=On \
+  -DCMAKE_C_COMPILER=clang \
+  -DCMAKE_CXX_COMPILER=clang++ \
+  -DCMAKE_LINKER=clang++ \
+  -DCMAKE_CXX_STANDARD=17 \
```
Comment (MrBurmark) [inline scripts/lc-builds/corona_sycl.sh:54]: What about BLT_CXX_STD?

Comment (rhornung67) [reply scripts/lc-builds/corona_sycl.sh:54]: I believe CMAKE_CXX_STANDARD covers it. I did not change this script, merely copied it over from RAJA repo so we can use/modify it when we have an updated sycl compiler to use.

**Review by MrBurmark — APPROVED** [review]

**Pull request merged**